lib/jekyll/commands/serve/servlet.rb

```diff
@@ -21,10 +21,43 @@ def initialize(server, root, callbacks)
         # up with a different preference on which comes first.
 
         def search_file(req, res, basename)
-          # /file.* > /file/index.html > /file.html
+          # /file.* -> /file.html
           super || super(req, res, "#{basename}.html")
         end
 
+        # rubocop:disable Lint/AssignmentInCondition
+        def search_index_file(req, res)
+          # /file/index.html -> /file.html
+
+          # First, let's see if the default implementation can figure it out.
+          # (Check for index files in the res.filename directory)
+          if file = super
+            return file
```

Comment:
I wasn't checking for equality. I was checking if the assigned value was truthy. if `file = super` evaluates to `nul` we continue, otherwise we return the value assigned to `file`.